lib/libzutil/zutil_import.c

```diff
@@ -1210,13 +1210,28 @@ label_paths(libpc_handle_t *hdl, nvlist_t *label, const char **path,
 	nvlist_t *nvroot;
 	uint64_t pool_guid;
 	uint64_t vdev_guid;
+	uint64_t state;
+	int is_aux = ((nvlist_lookup_uint64(label, ZPOOL_CONFIG_POOL_STATE,
+	    &state) == 0) && (state == POOL_STATE_SPARE ||
+	    state == POOL_STATE_L2CACHE));
 
 	*path = NULL;
 	*devid = NULL;
+	if (nvlist_lookup_uint64(label, ZPOOL_CONFIG_GUID, &vdev_guid) != 0)
+		return (ENOENT);
+
+	/*
+	 * In case of spare or l2cache, we directly return path/devid from the
+	 * label.
+	 */
+	if (is_aux) {
```

Comment:
I am not sure why do we need the is_aux variable, but whatever...

Reply:
Thank you for pointing it out. It was leftover and I removed it now.